### DISCUSSION TITLE
Cobblegen with bedrock always producing ores?

### DISCUSSION THREAD

**Discussion by Whattheblah12** [null2264/CobbleGen#14] [Q&A] [answered]
so im struggling to make a custom cobble generator for andesite granite and diorite and cant do it so im thinking hey maybe im doing something wrong, so i go to the config copy paste the andesite thing from the stone generator and replace the ores thing to see if itll start generating them on bedrock, then i could just change bedrock to what i want HOWEVER after saving the config and relaunching i have found that NO MATTER what is in the config, it will produce ores ontop of bedrock, and i cant make a custom gen (im using the newest version of curseforge and already doublechecked im in the right config) 

it did let me make a custom cobble gen for cobbled deepslate below y0 however so im not sure whats going on 
here is the code :
`        {
	/* Default Generators
	   {
	     "id": "mod_id:block_id",
	     "weight": 95.5,
	     "dimensions": [
	       "mod_id:dimension_id",
	       "mod_id:dimension_id"
	     ],
	     "excludedDimensions": [
	       "mod_id:dimension_id",
	       "mod_id:dimension_id"
	     ],
	     "minY": 0,
	     "maxY": 69
	   }
	*/
	"cobbleGen": [
		{
			"id": "minecraft:cobblestone",
			"weight": 100.0,
	
		},

	],
	"stoneGen": [
		{
			"id": "minecraft:stone",
			"weight": 100.0,
		},
		
	],
	"basaltGen": [
		{
			"id": "minecraft:basalt",
			"weight": 100.0,
		},
	],
	/* Custom Generators
	   <stoneGen|cobbleGen|basaltGen>: {
	     "mod_id:modifier_block_id": [
	       {
	         "id": "mod_id:block_id",
	         "weight": 95.5,
	         "dimensions": [
	           "mod_id:dimension_id",
	           "mod_id:dimension_id"
	         ],
	         "excludedDimensions": [
	           "mod_id:dimension_id",
	           "mod_id:dimension_id"
	         ],
	         "minY": 0,
	         "maxY": 69
	       },
	       ...
	     ]
	   }
	*/
	"customGen": {
		"cobbleGen": {
			"minecraft:bedrock": [
				{
					"id": "minecraft:cobblestone",
					"weight": 40.0,
				},
				{
					"id": "minecraft:diorite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:andesite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:granite",
					"weight": 20.0,
				},
			],
			"minecraft:andesite": [
				{
					"id": "minecraft:cobblestone",
					"weight": 40.0,
				},
				{
					"id": "minecraft:diorite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:andesite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:granite",
					"weight": 20.0,
				},
		},
		"cobbleGen": [
		{
			"id": "minecraft:cobbled_deepslate",
			"weight": 100.0,
			"maxY": 0
		},
		
		"basaltGen": {
			"minecraft:bedrock": [
				{
					"id": "minecraft:end_stone",
					"weight": 100.0,
					"dimensions": [
						"minecraft:the_end",
					],
				},
				{
					"id": "minecraft:blackstone",
					"weight": 100.0,
					"excludedDimensions": [
						"minecraft:overworld",
					],
				},
			],
		},
	},
}   `

this is what i get even after making a fresh world:
![image](https://user-images.githubusercontent.com/104280096/227756834-cc3c7979-91c1-4511-8b10-526911fa8fc7.png)

**Comment by null2264** [comment, marked as answer]
"customGen" is meant for custom generators that require a certain modifier block underneath the generated block, in this case, the default config requires players to have bedrock underneath where cobblestone is supposed to be generated to generate ores.

For "custom generators" that **don't** require any modifiers, you shouldn't put it inside "customGen". Also it seems like your config file is invalid, you should try using a validator/formatter like https://json-5.com/, because invalid configs will be ignored by the mod and it'll revert back to default config

```json5
{
	/* Default Generators
	   {
	     "id": "mod_id:block_id",
	     "weight": 95.5,
	     "dimensions": [
	       "mod_id:dimension_id",
	       "mod_id:dimension_id"
	     ],
	     "excludedDimensions": [
	       "mod_id:dimension_id",
	       "mod_id:dimension_id"
	     ],
	     "minY": 0,
	     "maxY": 69
	   }
	*/
/* --- THIS IS WHERE YOU SHOULD PUT IT --- */
	"cobbleGen": [
		{
			"id": "minecraft:cobblestone",
			"weight": 100.0,
		},
		{
			"id": "minecraft:cobbled_deepslate",
			"weight": 100.0,
			"maxY": 0
		},
	],
	"stoneGen": [
		{
			"id": "minecraft:stone",
			"weight": 100.0,
		},
		
	],
	"basaltGen": [
		{
			"id": "minecraft:basalt",
			"weight": 100.0,
		},
	],
	/* Custom Generators
	   <stoneGen|cobbleGen|basaltGen>: {
	     "mod_id:modifier_block_id": [
	       {
	         "id": "mod_id:block_id",
	         "weight": 95.5,
	         "dimensions": [
	           "mod_id:dimension_id",
	           "mod_id:dimension_id"
	         ],
	         "excludedDimensions": [
	           "mod_id:dimension_id",
	           "mod_id:dimension_id"
	         ],
	         "minY": 0,
	         "maxY": 69
	       },
	       ...
	     ]
	   }
	*/
	"customGen": {
		"cobbleGen": {
			"minecraft:bedrock": [
				{
					"id": "minecraft:cobblestone",
					"weight": 40.0,
				},
				{
					"id": "minecraft:diorite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:andesite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:granite",
					"weight": 20.0,
				},
			],
			"minecraft:andesite": [
				{
					"id": "minecraft:cobblestone",
					"weight": 40.0,
				},
				{
					"id": "minecraft:diorite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:andesite",
					"weight": 20.0,
				},
				{
					"id": "minecraft:granite",
					"weight": 20.0,
				},
			]
		},
		"basaltGen": {
			"minecraft:bedrock": [
				{
					"id": "minecraft:end_stone",
					"weight": 100.0,
					"dimensions": [
						"minecraft:the_end",
					],
				},
				{
					"id": "minecraft:blackstone",
					"weight": 100.0,
					"excludedDimensions": [
						"minecraft:overworld",
					],
				},
			],
		},
	},
}
```

**Reply by Whattheblah12** [reply]
if it was invalid why would the maxy cobblegen work? but ill try that thanks

**Reply by null2264** [reply]
Because that's the default configuration for cobblegen on the newest version of the mod.

**Reply by cwtuan** [reply]
Yes, it works after formatting json with comment via  https://json-5.com/
